libgalois/src/PropertyGraph.cpp

```diff
@@ -448,33 +564,25 @@ katana::PropertyGraph::Validate() {
   return katana::ResultSuccess();
 }
 
+/// Converts all uint8/bool properties into EntityTypeIDs
+/// Only call this if every uint8/bool property should be considered a type
 katana::Result<void>
 katana::PropertyGraph::ConstructEntityTypeIDs() {
-  node_entity_type_manager_.Reset();
-  uint64_t num_node_rows = static_cast<uint64_t>(node_properties()->num_rows());
-  if (num_node_rows == 0) {
-    node_entity_type_id_ = GetUnknownEntityTypeIDs(num_nodes());
-  } else {
-    auto node_types_res = GetEntityTypeIDsFromProperties(
-        node_properties(), &node_entity_type_manager_);
-    if (!node_types_res) {
-      return node_types_res.error().WithContext("node properties");
-    }
-    node_entity_type_id_ = std::move(node_types_res.value());
+  // only relevant to actually construct when EntityTypeIDs are expected in properties
+  // when EntityTypeIDs are not expected in properties then we have nothing to do here
+  if (rdg_.EntityTypeIDsOutsideProperties()) {
```

Comment:
This seems wrong. The initial file loaded from storage might be version 1 but this might be called after we update the graph in memory.